src/Core/Domain/Product/QueryResult/ProductForEditing.php

```diff
@@ -24,10 +24,15 @@
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  */
 
+declare(strict_types=1);
+
 namespace PrestaShop\PrestaShop\Core\Domain\Product\QueryResult;
 
 /**
  * Product information for editing
+ *
+ * @todo: should we follow {...}Information or {..}Options concept? do we need those suffixes at all?
```

Comment:
I think these names are valid when used in the right place.

1. For a product, you can enable multiple shipping options. Whether it's the weight, the carriers, the shipping fee ... it's settings ("options" and "settings" seem close) that merchant modifies to configure its sales flow.

2. However there are also "Informations" that merchant just puts there to be displayed to customer. Product description, product creation time, product expiration date ... it's "informations". They dont modify how the shop behaves they are just shown to customer.

In my opinion, suffixes are good because they carry an additional layer of informations to the developer. It reminds me of the debate "should we put `Interface` suffix for all interfaces or not ?

`ProductCategories` => what can I get from this name ? well, this is a variable that contain ... stuff about product and categories. Is it a relationship ? Is it references ? Is it informations ?

`ProductCategoriesAssociation` => aaaah a bit better

I'm from the "Java" school 😄  I like overdescriptive variable names. Better too much infos than not enough